test/e2e/specs/editor/various/writing-flow.spec.js

```diff
@@ -0,0 +1,1004 @@
+<figure class="wp-block-image alignwide"><img alt=""/></figure>
+<!-- /wp:image -->` );
+
+		const paragraphBlock = page.locator(
+			'role=document[name="Paragraph block"i]'
+		);
+
+		// Find a point outside the paragraph between the blocks where it's
+		// expected that the sibling inserter would be placed.
+		const paragraphRect = await paragraphBlock.boundingBox();
+		const x = paragraphRect.x + ( 2 * paragraphRect.width ) / 3;
+		const y = paragraphRect.y + paragraphRect.height + 1;
+
+		await page.mouse.move( x, y );
+
+		const inserter = page.locator( 'role=button[name="Add block"i]' );
+		await expect( inserter ).toBeVisible();
+
+		// Find the space between the inserter and the image block.
+		const inserterRect = await inserter.boundingBox();
+		const lowerInserterY = inserterRect.y + ( 2 * inserterRect.height ) / 3;
+
+		// Clicking that in-between space should select the image block.
+		await page.mouse.click( x, lowerInserterY );
+
+		await expect(
+			page.locator( 'role=document[name="Block: Image"i]' )
+		).toHaveClass( /is-selected/ );
+	} );
+
+	test( 'should only consider the content as one tab stop', async ( {
+		editor,
+		page,
+	} ) => {
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '/table' );
+		await page.keyboard.press( 'Enter' );
+		// Tab to the "Create table" button.
+		await page.keyboard.press( 'Tab' );
+		await page.keyboard.press( 'Tab' );
+		// Create the table.
+		await page.keyboard.press( 'Space' );
+		// Navigate to the second cell.
+		await page.keyboard.press( 'ArrowRight' );
+		await page.keyboard.type( '2' );
+		// Confirm correct setup.
+		await expect.poll( editor.getEditedPostContent )
+			.toBe( `<!-- wp:table -->
+<figure class="wp-block-table"><table><tbody><tr><td></td><td>2</td></tr><tr><td></td><td></td></tr></tbody></table></figure>
+<!-- /wp:table -->` );
+	} );
+
+	test( 'should unselect all blocks when hitting double escape', async ( {
+		page,
+		writingFlowUtils,
+	} ) => {
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'Random Paragraph' );
+
+		await expect
+			.poll( writingFlowUtils.getActiveBlockName )
+			.toBe( 'core/paragraph' );
+
+		// First escape enters navigaiton mode.
+		await page.keyboard.press( 'Escape' );
+		await expect
+			.poll( writingFlowUtils.getActiveBlockName )
+			.toBe( 'core/paragraph' );
+
+		// Second escape unselects the blocks.
+		await page.keyboard.press( 'Escape' );
+		await expect
+			.poll( writingFlowUtils.getActiveBlockName )
+			.toBe( undefined );
+	} );
+
+	// Checks for regressions of https://github.com/WordPress/gutenberg/issues/40091.
+	test( 'does not deselect the block when selecting text outside the editor canvas', async ( {
+		editor,
+		page,
+	} ) => {
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'Random Paragraph' );
+		await editor.openDocumentSettingsSidebar();
+		const blockDescription = page.locator(
+			'.block-editor-block-card__description'
+		);
```

Comment:
Off-topic: Maybe it's a sign that we want this element to be more accessible (`aria-describedby` or something similar).

Reply:
100%. I can look into this separately.